src/reactionsystem.jl

```diff
@@ -323,6 +323,62 @@ function MT.equations(sys::ReactionSystem)
     return eqs
 end
 
+function MT.defaults(sys::ReactionSystem)
+    constraints = get_constraints(sys)
+    defs = get_defaults(sys)
+    if constraints !== nothing
+        defs = merge(defs, MT.defaults(rename(constraints, nameof(sys))))
+    end
+    return defs
+end
+
+function MT.observed(sys::ReactionSystem)

```

Comment:
If you use `get_observed` on the constraint system they won't be namespaced, so can just be directly merged. I don't think we want to allow subsystems on the constraint system, so I don't think we need to worry about that.

For handling subsystems too, can you just mimic what is done at  

https://github.com/SciML/ModelingToolkit.jl/blob/ccf4af6336fb1fbabc51e960ea7b488cb054c47c/src/systems/abstractsystem.jl#L473-L480